mllib/src/main/scala/org/apache/spark/ml/tree/impl/GradientBoostedTrees.scala

```diff
@@ -329,8 +330,16 @@ private[spark] object GradientBoostedTrees extends Logging {
       //       However, the behavior should be reasonable, though not optimal.
       baseLearnerWeights(m) = learningRate
 
+      if (pre_predError.getStorageLevel != StorageLevel.NONE ){
```

Comment:
That's not quite what I mean -- it doesn't mean the RDD is done being evaluated, because it could occur in a lineage of RDDs. unpersisting here may simply make it not cache when it needs to be cached. There are similar patterns in other parts of the code.

But I'm still not clear how it relates to your problem?